src/Service/ViewHelperManagerFactory.php

```diff
@@ -5,13 +5,12 @@
 use Interop\Container\ContainerInterface;
 use Laminas\Router\RouteMatch;
 use Laminas\ServiceManager\Exception\ServiceNotCreatedException;
+use Laminas\ServiceManager\Factory\FactoryInterface;
 use Laminas\View\Helper as ViewHelper;
 use Laminas\View\HelperPluginManager;
 
-class ViewHelperManagerFactory extends AbstractPluginManagerFactory
+class ViewHelperManagerFactory implements FactoryInterface
```

Comment:
Honestly, I'd drop the `implements FactoryInterface` part. Factories only need to be invokable,and it's another explicit binding to laminas-servicemanager instead of a generic PSR-11 implementation.

Reply:
I agree.

Factories looks for extra input parameters that are only supported by service manager. the `$options` specifically, while for the `$name` we do provide decorators for others containers iirc.

I will have a pass dedicated to factories. I think quite a few of them can be safely dropped and replaced by factories provided by components.